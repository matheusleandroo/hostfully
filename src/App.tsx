import { ToastContainer } from 'react-toastify'
import { BookingsPage } from './pages/BookingsPage'

export default function App() {
  return (
    <>
      <ToastContainer />
      <BookingsPage />
    </>
  )
}
