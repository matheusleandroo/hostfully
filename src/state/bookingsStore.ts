import { create } from 'zustand'
import { persist } from 'zustand/middleware'
import { nanoid } from './nanoid'
import type { Booking } from '../lib/types'
import { overlaps, isInvalidRange } from '../lib/utils'

type State = {
  bookings: Booking[]
}
type Actions = {
  create: (
    b: Omit<Booking, 'id'>,
  ) => { ok: true; id: string } | { ok: false; reason: string }
  update: (
    id: string,
    b: Omit<Booking, 'id'>,
  ) => { ok: true } | { ok: false; reason: string }
  remove: (id: string) => void
  findById: (id: string) => Booking | undefined
}

export const useBookingsStore = create<State & Actions>()(
  persist(
    (set, get) => ({
      bookings: [],
      create: (b) => {
        if (isInvalidRange(b.startDate, b.endDate)) {
          return { ok: false, reason: 'End date must be after start date' }
        }
        const newBooking: Booking = { id: nanoid(), ...b }
        const conflict = get().bookings.some(
          (existing) =>
            existing.propertyName === b.propertyName &&
            overlaps(existing, newBooking),
        )
        if (conflict)
          return {
            ok: false,
            reason: 'Overlapping booking on the same property',
          }
        set((s) => ({ bookings: [newBooking, ...s.bookings] }))
        return { ok: true, id: newBooking.id }
      },
      update: (id, b) => {
        if (isInvalidRange(b.startDate, b.endDate)) {
          return { ok: false, reason: 'End date must be after start date' }
        }
        const updated: Booking = { id, ...b }
        const conflict = get().bookings.some(
          (existing) =>
            existing.id !== id &&
            existing.propertyName === b.propertyName &&
            overlaps(existing, updated),
        )
        if (conflict)
          return {
            ok: false,
            reason: 'Overlapping booking on the same property',
          }
        set((s) => ({
          bookings: s.bookings.map((x) => (x.id === id ? updated : x)),
        }))
        return { ok: true }
      },
      remove: (id) =>
        set((s) => ({ bookings: s.bookings.filter((x) => x.id !== id) })),
      findById: (id) => get().bookings.find((b) => b.id === id),
    }),
    { name: 'bookings-store' },
  ),
)
