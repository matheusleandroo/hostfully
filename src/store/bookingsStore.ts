import { create } from 'zustand'
import { persist } from 'zustand/middleware'
import { z } from 'zod'

export type Booking = {
  id: string
  guestName: string
  property: string
  notes?: string
  startDate: string
  endDate: string
  createdAt: string
  updatedAt: string
}

export type BookingInput = {
  guestName: string
  property: string
  notes?: string
  startDate: Date | null
  endDate: Date | null
  id?: string
}

type Result<T> = { ok: true; data: T } | { ok: false; error: string }

const bookingSchema = z.object({
  id: z.string(),
  guestName: z.string().min(1),
  property: z.string().min(1),
  notes: z.string().optional(),
  startDate: z.string().datetime(),
  endDate: z.string().datetime(),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
})

const bookingsSchema = z.array(bookingSchema)

function rangesOverlap(aStart: Date, aEnd: Date, bStart: Date, bEnd: Date) {
  return aStart <= bEnd && bStart <= aEnd
}

function normalizeStr(v: string) {
  return (v ?? '').trim().replace(/\s+/g, ' ')
}

type State = {
  items: Booking[]
  isLoading: boolean
  error: string | null
  create: (input: BookingInput) => Promise<Result<Booking>>
  update: (id: string, input: BookingInput) => Promise<Result<Booking>>
  remove: (id: string) => Promise<Result<string>>
  clearError: () => void
}

const STORAGE_KEY = 'hostfully_bookings_v1'

export const useBookingsStore = create<State>()(
  persist(
    (set, get) => ({
      items: [],
      isLoading: false,
      error: null,

      clearError: () => set({ error: null }),

      async create(input) {
        set({ isLoading: true, error: null })
        try {
          const validation = validateInput(input)
          if (!validation.ok) return fail(validation.error)

          const now = new Date()
          const id =
            globalThis.crypto?.randomUUID?.() ??
            `${now.getTime()}-${Math.random().toString(16).slice(2)}`
          const newBooking: Booking = {
            id,
            guestName: normalizeStr(input.guestName),
            property: normalizeStr(input.property),
            notes: input.notes?.trim() || '',
            startDate: validation.data.start.toISOString(),
            endDate: validation.data.end.toISOString(),
            createdAt: now.toISOString(),
            updatedAt: now.toISOString(),
          }

          const overlap = get().items.some(
            (b) =>
              b.property === newBooking.property &&
              rangesOverlap(
                new Date(b.startDate),
                new Date(b.endDate),
                validation.data.start,
                validation.data.end,
              ),
          )
          if (overlap)
            return fail(
              'This date range overlaps an existing booking for this property.',
            )

          const items = [newBooking, ...get().items]
          set({ items, isLoading: false })
          return ok(newBooking)
        } catch {
          set({ isLoading: false, error: 'Failed to create booking.' })
          return fail('Failed to create booking.')
        }
      },

      async update(id, input) {
        set({ isLoading: true, error: null })
        try {
          const existing = get().items.find((b) => b.id === id)
          if (!existing) return fail('Booking not found.')

          const validation = validateInput(input)
          if (!validation.ok) return fail(validation.error)

          const updated: Booking = {
            ...existing,
            guestName: normalizeStr(input.guestName),
            property: normalizeStr(input.property),
            notes: input.notes?.trim() || '',
            startDate: validation.data.start.toISOString(),
            endDate: validation.data.end.toISOString(),
            updatedAt: new Date().toISOString(),
          }

          const overlap = get().items.some(
            (b) =>
              b.id !== id &&
              b.property === updated.property &&
              rangesOverlap(
                new Date(b.startDate),
                new Date(b.endDate),
                new Date(updated.startDate),
                new Date(updated.endDate),
              ),
          )
          if (overlap)
            return fail(
              'This date range overlaps an existing booking for this property.',
            )

          const items = get().items.map((b) => (b.id === id ? updated : b))
          set({ items, isLoading: false })
          return ok(updated)
        } catch {
          set({ isLoading: false, error: 'Failed to update booking.' })
          return fail('Failed to update booking.')
        }
      },

      async remove(id) {
        set({ isLoading: true, error: null })
        try {
          const exists = get().items.some((b) => b.id === id)
          if (!exists) return fail('Booking not found.')

          set({
            items: get().items.filter((b) => b.id !== id),
            isLoading: false,
          })
          return ok(id)
        } catch {
          set({ isLoading: false, error: 'Failed to delete booking.' })
          return fail('Failed to delete booking.')
        }
      },
    }),
    {
      name: STORAGE_KEY,
      version: 1,
      partialize: (state) => ({ items: state.items }),
      onRehydrateStorage: () => (state) => {
        try {
          const raw = (state as any)?.items
          const parsed = bookingsSchema.safeParse(raw)
          if (!parsed.success) {
            ;(state as any).items = []
          }
        } catch {
          // ignore
        }
      },
    },
  ),
)

function ok<T>(data: T): Result<T> {
  return { ok: true, data }
}

function fail<T = never>(error: string): Result<T> {
  return { ok: false, error }
}

function validateInput(
  input: BookingInput,
): Result<{ start: Date; end: Date }> {
  const guest = normalizeStr(input.guestName || '')
  const property = normalizeStr(input.property || '')
  if (!guest) return fail('Guest name is required.')
  if (!property) return fail('Property is required.')
  if (!input.startDate || !input.endDate)
    return fail('Start and end dates are required.')

  const start = new Date(input.startDate)
  const end = new Date(input.endDate)
  if (Number.isNaN(start.getTime()) || Number.isNaN(end.getTime()))
    return fail('Invalid dates.')
  if (end < start) return fail('End date cannot be before start date.')

  const MAX_RANGE_DAYS = 365
  const diffDays = Math.ceil((+end - +start) / 86_400_000) + 1
  if (diffDays > MAX_RANGE_DAYS) return fail('Date range is too long.')

  return ok({ start, end })
}
