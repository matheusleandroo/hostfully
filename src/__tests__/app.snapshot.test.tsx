import { render } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import App from '../App'
import { useBookingsStore } from '../store/bookingsStore'

describe('App snapshot', () => {
  it('matches', () => {
    const { getState, setState } = useBookingsStore
    setState({ ...getState(), items: [], isLoading: false, error: null }, true)
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
