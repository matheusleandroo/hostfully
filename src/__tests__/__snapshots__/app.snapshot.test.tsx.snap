// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App snapshot > matches 1`] = `
<div>
  <section
    aria-atomic="false"
    aria-label="Notifications Alt+T"
    aria-live="polite"
    aria-relevant="additions text"
    class="Toastify"
  />
  <div
    class="max-w-5xl mx-auto p-4 md:p-8 grid gap-6"
  >
    <header
      class="flex flex-col md:flex-row md:items-center md:justify-between gap-3"
    >
      <h1
        class="text-2xl md:text-3xl font-bold tracking-tight"
      >
        Bookings Manager
      </h1>
      <p
        class="text-slate-600"
      >
        Create, read, update and delete bookings with validation & overlaps prevention.
      </p>
    </header>
    <section>
      <form
        class="card grid gap-4"
      >
        <h2
          class="text-xl font-bold"
        >
          Booking 
          Create
        </h2>
        <div
          class="grid md:grid-cols-2 gap-4"
        >
          <label
            class="grid gap-1"
          >
            <span
              class="label"
            >
              Guest name *
            </span>
            <input
              class="input"
              placeholder="John Doe"
              required=""
              value=""
            />
          </label>
          <label
            class="grid gap-1"
          >
            <span
              class="label"
            >
              Property *
            </span>
            <input
              class="input"
              placeholder="Beach House"
              required=""
              value=""
            />
          </label>
        </div>
        <div
          class="grid gap-2"
        >
          <span
            class="label"
          >
            Dates
          </span>
          <div
            class="card w-[fit-content]"
          >
            <div
              class="rdp-root"
              data-mode="range"
            >
              <div
                class="rdp-months"
              >
                <nav
                  aria-label=""
                  class="rdp-nav"
                >
                  <button
                    aria-label="Go to the Previous Month"
                    class="rdp-button_previous"
                    type="button"
                  >
                    <svg
                      class="rdp-chevron"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <polygon
                        points="16 18.112 9.81111111 12 16 5.87733333 14.0888889 4 6 12 14.0888889 20"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Go to the Next Month"
                    class="rdp-button_next"
                    type="button"
                  >
                    <svg
                      class="rdp-chevron"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <polygon
                        points="8 18.112 14.18888889 12 8 5.87733333 9.91111111 4 18 12 9.91111111 20"
                      />
                    </svg>
                  </button>
                </nav>
                <div
                  class="rdp-month"
                >
                  <div
                    class="rdp-month_caption"
                  >
                    <span
                      aria-live="polite"
                      class="rdp-caption_label"
                      role="status"
                    >
                      October 2025
                    </span>
                  </div>
                  <table
                    aria-label="October 2025"
                    aria-multiselectable="true"
                    class="rdp-month_grid"
                    role="grid"
                  >
                    <thead
                      aria-hidden="true"
                    >
                      <tr
                        class="rdp-weekdays"
                      >
                        <th
                          aria-label="Sunday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Su
                        </th>
                        <th
                          aria-label="Monday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Mo
                        </th>
                        <th
                          aria-label="Tuesday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Tu
                        </th>
                        <th
                          aria-label="Wednesday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          We
                        </th>
                        <th
                          aria-label="Thursday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Th
                        </th>
                        <th
                          aria-label="Friday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Fr
                        </th>
                        <th
                          aria-label="Saturday"
                          class="rdp-weekday"
                          scope="col"
                        >
                          Sa
                        </th>
                      </tr>
                    </thead>
                    <tbody
                      class="rdp-weeks"
                    >
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2025-09-28"
                          data-hidden="true"
                          data-month="2025-09"
                          data-outside="true"
                          role="gridcell"
                        />
                        <td
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2025-09-29"
                          data-hidden="true"
                          data-month="2025-09"
                          data-outside="true"
                          role="gridcell"
                        />
                        <td
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2025-09-30"
                          data-hidden="true"
                          data-month="2025-09"
                          data-outside="true"
                          role="gridcell"
                        />
                        <td
                          class="rdp-day"
                          data-day="2025-10-01"
                          role="gridcell"
                        >
                          <button
                            aria-label="Wednesday, October 1st, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            1
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-02"
                          role="gridcell"
                        >
                          <button
                            aria-label="Thursday, October 2nd, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            2
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-03"
                          role="gridcell"
                        >
                          <button
                            aria-label="Friday, October 3rd, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            3
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-04"
                          role="gridcell"
                        >
                          <button
                            aria-label="Saturday, October 4th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            4
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day"
                          data-day="2025-10-05"
                          role="gridcell"
                        >
                          <button
                            aria-label="Sunday, October 5th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            5
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-06"
                          role="gridcell"
                        >
                          <button
                            aria-label="Monday, October 6th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            6
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-07"
                          role="gridcell"
                        >
                          <button
                            aria-label="Tuesday, October 7th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            7
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-08"
                          role="gridcell"
                        >
                          <button
                            aria-label="Wednesday, October 8th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            8
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-09"
                          role="gridcell"
                        >
                          <button
                            aria-label="Thursday, October 9th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            9
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-10"
                          role="gridcell"
                        >
                          <button
                            aria-label="Friday, October 10th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            10
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-11"
                          role="gridcell"
                        >
                          <button
                            aria-label="Saturday, October 11th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            11
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day"
                          data-day="2025-10-12"
                          role="gridcell"
                        >
                          <button
                            aria-label="Sunday, October 12th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            12
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-13"
                          role="gridcell"
                        >
                          <button
                            aria-label="Monday, October 13th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            13
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-14"
                          role="gridcell"
                        >
                          <button
                            aria-label="Tuesday, October 14th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            14
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-15"
                          role="gridcell"
                        >
                          <button
                            aria-label="Wednesday, October 15th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            15
                          </button>
                        </td>
                        <td
                          aria-selected="true"
                          class="rdp-day rdp-today rdp-selected rdp-range_start"
                          data-day="2025-10-16"
                          data-selected="true"
                          data-today="true"
                          role="gridcell"
                        >
                          <button
                            aria-label="Today, Thursday, October 16th, 2025, selected"
                            class="rdp-day_button"
                            tabindex="0"
                            type="button"
                          >
                            16
                          </button>
                        </td>
                        <td
                          aria-selected="true"
                          class="rdp-day rdp-selected rdp-range_end"
                          data-day="2025-10-17"
                          data-selected="true"
                          role="gridcell"
                        >
                          <button
                            aria-label="Friday, October 17th, 2025, selected"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            17
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-18"
                          role="gridcell"
                        >
                          <button
                            aria-label="Saturday, October 18th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            18
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day"
                          data-day="2025-10-19"
                          role="gridcell"
                        >
                          <button
                            aria-label="Sunday, October 19th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            19
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-20"
                          role="gridcell"
                        >
                          <button
                            aria-label="Monday, October 20th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            20
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-21"
                          role="gridcell"
                        >
                          <button
                            aria-label="Tuesday, October 21st, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            21
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-22"
                          role="gridcell"
                        >
                          <button
                            aria-label="Wednesday, October 22nd, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            22
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-23"
                          role="gridcell"
                        >
                          <button
                            aria-label="Thursday, October 23rd, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            23
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-24"
                          role="gridcell"
                        >
                          <button
                            aria-label="Friday, October 24th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            24
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-25"
                          role="gridcell"
                        >
                          <button
                            aria-label="Saturday, October 25th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            25
                          </button>
                        </td>
                      </tr>
                      <tr
                        class="rdp-week"
                      >
                        <td
                          class="rdp-day"
                          data-day="2025-10-26"
                          role="gridcell"
                        >
                          <button
                            aria-label="Sunday, October 26th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            26
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-27"
                          role="gridcell"
                        >
                          <button
                            aria-label="Monday, October 27th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            27
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-28"
                          role="gridcell"
                        >
                          <button
                            aria-label="Tuesday, October 28th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            28
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-29"
                          role="gridcell"
                        >
                          <button
                            aria-label="Wednesday, October 29th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            29
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-30"
                          role="gridcell"
                        >
                          <button
                            aria-label="Thursday, October 30th, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            30
                          </button>
                        </td>
                        <td
                          class="rdp-day"
                          data-day="2025-10-31"
                          role="gridcell"
                        >
                          <button
                            aria-label="Friday, October 31st, 2025"
                            class="rdp-day_button"
                            tabindex="-1"
                            type="button"
                          >
                            31
                          </button>
                        </td>
                        <td
                          class="rdp-day rdp-hidden rdp-outside"
                          data-day="2025-11-01"
                          data-hidden="true"
                          data-month="2025-11"
                          data-outside="true"
                          role="gridcell"
                        />
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
            <div
              class="text-sm text-slate-600 mt-2 flex gap-2"
            >
              <span>
                Start: 
                Oct 16, 2025
              </span>
              <span>
                End: 
                Oct 17, 2025
              </span>
            </div>
          </div>
        </div>
        <label
          class="grid gap-1"
        >
          <span
            class="label"
          >
            Notes
          </span>
          <textarea
            class="input min-h-24"
            placeholder="Optional notes..."
          />
        </label>
        <div
          class="flex gap-2 justify-end"
        >
          <button
            class="btn-primary"
            type="submit"
          >
            Create booking
          </button>
        </div>
      </form>
    </section>
    <section
      class="grid gap-3"
    >
      <h2
        class="text-xl font-semibold"
      >
        All bookings
      </h2>
      <p
        class="text-center text-slate-500 mt-8"
      >
        No bookings yet. Create your first one above.
      </p>
    </section>
    <footer
      class="text-center text-xs text-slate-500 py-6"
    >
      Built with love by
       
      <a
        class="text-cyan-900"
        href="https://matheusleandro.com/"
        target="_blank"
      >
        Matheus
      </a>
      .
    </footer>
  </div>
</div>
`;
